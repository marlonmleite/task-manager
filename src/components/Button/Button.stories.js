import React from 'react'
import { storiesOf } from '@storybook/react'
import Button from './'

const storie = storiesOf('Button', module)

storie.add('primary', () => <Button color="primary">My Button</Button>)
