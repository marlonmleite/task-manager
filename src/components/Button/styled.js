import styled from 'styled-components'

export const ButtonStyled = styled.button`
  padding: 0.4375rem 0.75rem;
  line-height: 1.25rem;
  font-size: 0.875rem;
  background-color: ${({ theme, color }) => theme.colors[color]};
  color: ${({ theme, color }) => theme.button.text[color]};
  display: inline-block;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  user-select: none;
  border: 1px solid ${({ theme, color }) => theme.colors[color]};
  border-radius: 2px;
  position: relative;

  :focus,
  :hover {
    text-decoration: none;
    background-image: linear-gradient(transparent, rgba(0,0,0,.05) 40%, rgba(0,0,0,.1));
  }

  :focus {
    outline: none;
  }

  .disabled,
  :disabled {
    opacity: .65;
    box-shadow: none;
  }

  :not([disabled]):not(.disabled) {
    cursor: pointer;
  }

  ${({ loading }) => loading && `
    > span {
      visibility: hidden;
    }
  `};
`

export const LoadingContainer = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
`
