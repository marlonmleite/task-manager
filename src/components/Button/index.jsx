import PropTypes from 'prop-types'
import React, { PureComponent } from 'react'
import { Loading } from 'components/Loading'
import { ButtonStyled, LoadingContainer } from './styled'

class Button extends PureComponent {

  static defaultProps = {
    color: 'primary',
    type: 'button',
    onClick: null,
    loading: false,
  }

  static propTypes = {
    children: PropTypes.node.isRequired,
    onClick: PropTypes.func,
    color: PropTypes.string,
    type: PropTypes.string,
    loading: PropTypes.bool,
  }

  render() {
    const { children, ...props } = this.props

    return (
      <ButtonStyled {...props}>
        <span>{children}</span>
        {props.loading && (
          <LoadingContainer>
            <Loading size="14px" color="secondary" />
          </LoadingContainer>
        )}
      </ButtonStyled>
    )
  }

}

export default Button
