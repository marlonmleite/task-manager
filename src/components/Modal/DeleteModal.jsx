import PropTypes from 'prop-types'
import React, { Component } from 'react'
import get from 'lodash/get'
import Button from '../Button'
import { Modal } from './'
import { ModalFooter, ModalBody, ModalContent } from './styled'

class DeleteModal extends Component {

  static defaultProps = {
    item: null,
    labelKey: 'description',
  }

  static propTypes = {
    item: PropTypes.object,
    onClose: PropTypes.func.isRequired,
    onConfirm: PropTypes.func.isRequired,
    labelKey: PropTypes.string,
  }

  onConfirm = () => {
    const { onConfirm, onClose, item } = this.props

    onConfirm(item)
    onClose()
  }

  render() {
    const { item, onClose, labelKey } = this.props

    return (
      <Modal onClose={onClose} open center>
        <ModalBody>
          <ModalContent>
            <p>Você deseja remover <b>{get(item, labelKey)}</b>?</p>
          </ModalContent>
          <ModalFooter>
            <Button color="secondary" onClick={this.onConfirm}>Sim</Button>
            <Button color="secondary" onClick={onClose}>Não</Button>
          </ModalFooter>
        </ModalBody>
      </Modal>
    )
  }

}

export default DeleteModal
