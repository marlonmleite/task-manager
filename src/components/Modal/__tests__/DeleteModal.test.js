import React from 'react'
import { shallow } from 'enzyme'
import Button from '../../Button'
import DeleteModal from '../DeleteModal'

describe('DeleteModal', () => {
  it('should render', () => {
    const modal = shallow(<DeleteModal onClose={jest.fn()} onConfirm={jest.fn()} />)

    expect(modal).toMatchSnapshot()
  })

  it('should close modal', () => {
    const onConfirm = jest.fn()
    const onClose = jest.fn()
    const modal = shallow(<DeleteModal onClose={onClose} onConfirm={onConfirm} />)

    modal.find(Button).last().simulate('click')

    expect(onConfirm).not.toHaveBeenCalled()
    expect(onClose).toHaveBeenCalled()
  })

  it('should confirm and close modal', () => {
    const onConfirm = jest.fn()
    const onClose = jest.fn()
    const modal = shallow(<DeleteModal onClose={onClose} onConfirm={onConfirm} />)

    modal.find(Button).first().simulate('click')

    expect(onConfirm).toHaveBeenCalled()
    expect(onClose).toHaveBeenCalled()
  })
})
