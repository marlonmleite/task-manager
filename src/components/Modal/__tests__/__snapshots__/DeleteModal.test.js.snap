// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteModal should render 1`] = `
<Modal
  animationDuration={500}
  blockScroll={true}
  center={true}
  classNames={Object {}}
  classes={
    Object {
      "closeButton": "styles_closeButton__20ID4",
      "closeIcon": "styles_closeIcon__1QwbI",
      "modal": "styles_modal__gNwvD",
      "overlay": "styles_overlay__CLSq-",
      "overlayCenter": "styles_overlayCenter__YHoO7",
      "transitionEnter": "styles_transitionEnter__3j_-a",
      "transitionEnterActive": "styles_transitionEnterActive___eQs7",
      "transitionExit": "styles_transitionExit__1KmEf",
      "transitionExitActive": "styles_transitionExitActive__1nQXw",
    }
  }
  closeIconSize={28}
  closeIconSvgPath={
    <path
      d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
    />
  }
  closeOnEsc={true}
  closeOnOverlayClick={true}
  onClose={[MockFunction]}
  onEntered={null}
  onEscKeyDown={null}
  onExited={null}
  onOverlayClick={null}
  open={true}
  showCloseIcon={true}
  styles={Object {}}
>
  <styled.div>
    <styled.div
      size="auto"
    >
      <p>
        Você deseja remover 
        <b />
        ?
      </p>
    </styled.div>
    <styled.div
      justify="center"
    >
      <Button
        color="secondary"
        loading={false}
        onClick={[Function]}
        type="button"
      >
        Sim
      </Button>
      <Button
        color="secondary"
        loading={false}
        onClick={[MockFunction]}
        type="button"
      >
        Não
      </Button>
    </styled.div>
  </styled.div>
</Modal>
`;
